plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -14,25 +15,37 @@ type TeamCache<T> = Map<TeamId, [T, number]>
 export class TeamManager {
     db: DB
     teamCache: TeamCache<Team | null>
-    eventNamesCache: Map<TeamId, Set<string>>
-    eventLastSeenCache: Map<string, number> // key: ${team_id}_${name}; value: DateTime.valueOf()
+    eventDefinitionsCache: Map<TeamId, Set<string>>
+    eventPropertiesCache: LRU<string, Set<string>> // Map<JSON.stringify([TeamId, Event], Set<Property>>
+    eventLastSeenCache: Map<string, number> // key: JSON.stringify([team_id, event]); value: timestamp (in ms)
     lastFlushAt: DateTime // time when the `eventLastSeenCache` was last flushed
-    eventPropertiesCache: Map<TeamId, Set<string>>
+    propertyDefinitionsCache: Map<TeamId, Set<string>>
     instanceSiteUrl: string
     experimentalLastSeenAtEnabled: boolean
+    propertyCounterTeams: Set<number>
     statsd?: StatsD
 
-    // TODO: #7422 Remove temporary parameter
-    constructor(db: DB, statsd?: StatsD, instanceSiteUrl?: string | null, experimentalLastSeenAtEnabled?: boolean) {
+    constructor(db: DB, serverConfig: PluginsServerConfig, statsd?: StatsD) {
         this.db = db
         this.statsd = statsd
         this.teamCache = new Map()
-        this.eventNamesCache = new Map()
+        this.eventDefinitionsCache = new Map()
+        this.eventPropertiesCache = new LRU({
+            max: serverConfig.EVENT_PROPERTY_LRU_SIZE, // keep in memory the last 10k team+event combos we have seen
+            maxAge: 60 * 60 * 1000, // and each for up to 1 hour
```

Comment:
Worth extracting to constants.ts given it's used in another file as well?

Reply:
Hmm... I guess we can. Moved `ONE_HOUR` to a new file called `constants.ts`, and imported in the 4 places it's used.

While all 4 places do reference the time diff of 60*60*1000 milliseconds, they're not connected, and having them all use the same const feels funny. Ideally the constants could be called something like `EVENT_PROPERTY_LRU_MAX_AGE`... and then moved out of `contants.ts` into `config.ts` with the other envs... but I'm going to draw a line here.